public/app/features/datasources/DataSourcesListItem.tsx

```diff
@@ -1,32 +1,123 @@
+  }
+
+  onHoverEnter = () => {
+    this.setState({ hoverOpen: true });
+  };
+
+  onHoverLeave = () => {
+    this.setState({ hoverOpen: false });
+  };
+
+  toggleDeleteOpen = () => {
+    this.setState({
+      deleteConfirmOpen: !this.state.deleteConfirmOpen,
+    });
+  };
+
+  onDelete = () => {
+    appEvents.emit(CoreEvents.showConfirmModal, {
+      title: 'Delete',
+      text: 'Are you sure you want to delete this data source?',
+      yesText: 'Delete',
+      icon: 'fa-trash',
+      onConfirm: () => {
+        this.props.deleteDataSource(this.props.dataSource.id);
+      },
+    });
+  };
+
   render() {
     const { dataSource } = this.props;
+
     return (
-      <li className="card-item-wrapper">
-        <a className="card-item" href={`datasources/edit/${dataSource.id}`}>
-          <div className="card-item-header">
-            <div className="card-item-type">{dataSource.type}</div>
-          </div>
-          <div className="card-item-body">
-            <figure className="card-item-figure">
-              <img src={dataSource.typeLogoUrl} alt={dataSource.name} />
-            </figure>
-            <div className="card-item-details">
-              <div className="card-item-name">
-                {dataSource.name}
-                {dataSource.isDefault && <span className="btn btn-secondary btn-small card-item-label">default</span>}
+      <li
+        className="card-item-wrapper configuration-card-item"
+        onMouseEnter={this.onHoverEnter}
+        onMouseLeave={this.onHoverLeave}
+      >
+        <div className="card-item">
+          <div className="configuration-card-item-wrapper">
+            <div className="configuration-card-item-column configuration-card-item-column--logo">
+              <figure className="card-item-figure">
+                <img src={dataSource.typeLogoUrl} alt={dataSource.name} />
+              </figure>
+            </div>
+            <div className="configuration-card-item-column configuration-card-item-column--content">
+              <div className="configuration-card-item-column configuration-card-item-column--left">
+                <div className="card-item-details">
+                  <div className="card-item-name configuration-card-item-name">
+                    {dataSource.name}
+                    {dataSource.isDefault && (
+                      <span className="btn btn-secondary btn-small card-item-label">default</span>
+                    )}
+                  </div>
+                  <div className="card-item-sub-name">{dataSource.type}</div>
+                </div>
+              </div>
+              <div className="configuration-card-item-column configuration-card-item-column--center">
+                <div className="card-item-name">{dataSource.url}</div>
+              </div>
+
+              <div className="configuration-card-item-column configuration-card-item-column--right">
+                {!this.state.deleteConfirmOpen && this.state.hoverOpen ? (
```

Comment:
does this hover state hide/show need to be done in js? Seems like a simple css show on hover could be simpler and remove the onMouseEventer / onMouseLeave handlers.